### PR TITLE
Docs v2

### PR DESCRIPTION
**Pull request by skryukov**
This PR adopts https://v2.inertiajs.com docs for the Inertia Rails plugin.

See also: [Inertia.js v2.0 compatibility PR](https://github.com/inertiajs/inertia-rails/pull/132)

### CODE REVIEW

File: docs/guide/partial-reloads.md [hunk 1]
```diff
@@ -203,13 +164,42 @@ end
...
 
 ```ruby
 class UsersController < ApplicationController
   def index
     render inertia: 'Users/Index', props: {
-      users: InertiaRails.lazy(-> { User.all }),
+      users: InertiaRails.always(User.all),
+
+      # Also works with block:
+      # users: InertiaRails.optional { User.all },
```
Comment (mpressen) [inline docs/guide/partial-reloads.md:199]:
```suggestion
      # users: InertiaRails.always { User.all },
```

File: docs/guide/partial-reloads.md [hunk 2]
```diff
@@ -203,13 +164,42 @@ end
...
+
+      # Also works with a lambda:
+      # users: InertiaRails.optional(-> { User.all }),
```
Comment (mpressen) [inline docs/guide/partial-reloads.md:202]:
```suggestion
      # users: InertiaRails.always(-> { User.all }),
```

File: docs/guide/polling.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+If you need to pass additional request options to the poll helper, you can pass any of the `router.reload` options as the second parameter.
+
+
+:::tabs key:frameworks
+== Vue
+
+```js
+import { usePoll } from '@inertiajs/vue3'
+
+usePoll(2000, {
```
Comment (mpressen) [inline docs/guide/polling.md:41]:
```suggestion
usePoll(2000, {}, {
```
second argument is a shortcut for data right ?

File: docs/guide/polling.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+```
+
+== React
+
+```jsx
+import { usePoll } from '@inertiajs/react'
+
+export default () => {
+  const { start, stop } = usePoll(2000, {}, {
```
Comment (mpressen) [inline docs/guide/polling.md:108]:
```suggestion
  const { start, stop } = usePoll(2000, {
```
As said `If you need to pass additional request options to the poll helper, you can pass any of the router.reload options as the second parameter.`

and in Manual visits : `router.reload(options)`.

No data as a second argument, unlike get/post/put/patch

File: docs/guide/polling.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+}
+```
+
+== Svelte 4|Svelte 5
+
+```svelte
+<script>
+import { usePoll } from '@inertiajs/svelte'
+
+const { start, stop } = usePoll(2000, {}, {
```
Comment (mpressen) [inline docs/guide/polling.md:126]:
```suggestion
const { start, stop } = usePoll(2000, {
```

File: docs/guide/polling.md [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+If you'd like more control over the polling behavior, the poll helper provides `stop` and  `start` methods that allow you to manually start and stop polling. You can pass the `autoStart: false` option to the poll helper to prevent it from automatically starting polling when the component is mounted.
+
+:::tabs key:frameworks
+== Vue
+
+```vue
+<script setup>
+import { usePoll } from '@inertiajs/vue3'
+const { start, stop } = usePoll(2000, {}, {
```
Comment (mpressen) [inline docs/guide/polling.md:90]:
```suggestion
const { start, stop } = usePoll(2000, {
```

File: docs/guide/polling.md (hunk 4 above)
Comment (skryukov) [reply docs/guide/polling.md:108]: https://github.com/inertiajs/inertia/blame/9106d74495130a3d82d00daec43c4042b81bdb63/packages/react/src/usePoll.ts#L7 I believe this one is fine, but tell me if I'm missing smth 👀

File: docs/guide/load-when-visible.md [hunk 7]
```diff
@@ -0,0 +1,349 @@
...
+<WhenVisible data="permissions" buffer={500}>
+  <svelte:fragment slot="fallback">
+    <div>Loading...</div>
+  </svelte:fragment>
+  
+  {#each permissions as permission}
+    <!-- ... -->
+  {/each}
+</WhenVisible>
+
```
Comment (mpressen) [inline docs/guide/load-when-visible.md:205]: the end of the svelte code block for svelte 4 is missing. Therefore the text explaining the 'as' appears only when selecting svelte 5.
image
image

File: docs/guide/polling.md (hunk 4 above)
Comment (mpressen) [reply docs/guide/polling.md:108]: You're right, sorry for the inconvenience !

**Comment by skryukov** [comment]
Close in favor of https://github.com/inertiajs/inertia-rails/pull/148

**Pull request closed without merging**